packages/expo-facebook/android/src/main/java/expo/modules/facebook/FacebookModule.java

```diff
@@ -124,14 +103,27 @@ public void onError(FacebookException error) {
     });
 
     try {
-      LoginManager.getInstance().logInWithReadPermissions(Exponent.getInstance().getCurrentActivity(), permissions);
+      LoginManager.getInstance().logInWithReadPermissions(mModuleRegistry.getModule(ActivityProvider.class).getCurrentActivity(), permissions);
     } catch (FacebookException e) {
       promise.reject("E_FBLOGIN_ERROR", "An error occurred while trying to log in to Facebook", e);
     }
   }
 
   @Override
-  public void onActivityResult(final int requestCode, final int resultCode, final Intent data) {
+  public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+    mModuleRegistry = moduleRegistry;
+    if (mModuleRegistry != null) {
+      mModuleRegistry.getModule(UIManager.class).registerActivityEventListener(this);
```

Comment:
Please remind me once we merge my PR #3273 (that implements `unregisterActivityEventListener`) to unregister the listener as we do for lifecycle listeners.